sydent/server.py

```diff
@@ -0,0 +1,81 @@
+from zope.interface import implementer
+
+from twisted.internet import udp
+
+from twisted.python.failure import Failure
+from twisted.internet._resolver import SimpleResolverComplexifier
+from twisted.internet.defer import Deferred, fail, succeed
+from twisted.internet.error import DNSLookupError
+from twisted.internet.interfaces import (
+    IReactorPluggableNameResolver,
+    IResolverSimple,
+)
+from twisted.test.proto_helpers import MemoryReactorClock
+
+@implementer(IReactorPluggableNameResolver)
+class ThreadedMemoryReactorClock(MemoryReactorClock):
+    """
+    A MemoryReactorClock that supports callFromThread.
+    """
+
+    def __init__(self):
+        self.threadpool = ThreadPool(self)
+
+        self._udp = []
+        lookups = self.lookups = {}
```

Comment:
Could you please document what is the use for these variables, why we need two of them, and how they are populated? It seems quite unclear to me.

Reply:
This was copy/pasted from Synapse, hm

Reply:
Well, the first thing to note is that it's not actually used here.

not that it's not really two variables: it's two names for the same dict.

The intention is that it is populated by a test, which might do something like:

```
        self.lookups["matrix.org"] = [(IPv4Address, "8.8.8.8")]
```

which would them mean a DNS lookup for `matrix.org` later in the test would return 8.8.8.8. So that's how it's meant to be populated, and why it needs to be accessible from `self`.

it also needs to be accessible via a local var, because line 30 has a different `self`, so can't reach it via `self.lookups`.

Reply:
Will add the appropriate comment (to both repos). Thanks!